src/mixxx.cpp

```diff
@@ -598,6 +596,11 @@ void MixxxMainWindow::initializeWindow() {
     Pal.setColor(QPalette::Background, MenuBarBackground);
     m_pMenuBar->setPalette(Pal);
 
+    // restore window size/position
+    QSettings settings("mixxxdj", "mixxx");
```

Comment:
> Why would you need to do anything with the widget states after changing to a new profile format? If they become invalid for some reason then just delete them. Deletion is completely acceptable for geometry state, and most other widget states that I can think of storing.

If users can't trust that their settings will be saved and that all upgrades are backwards compatible then they will lose faith in Mixxx's technical design. Since DJs bet their careers on Mixxx, it's important to us that all user state that is stored is preserved.

> If you still insist on storing the widget state in the config file, you'll need to be more explicit about exactly how you want to encode the byte arrays returned by saveGeometry() and saveState()

I do insist that you put in some effort to make this more maintainable. I can understand why it's tempting to drive straight from point A to point B but sometimes we have to drive on the pre-paved roads to keep things orderly.

My previous example implicitly expressed this, but I'd like you store the widget state base64 encoded in the config file. Since it's binary data and mixxx.cfg is a plain text file base64 encoding will allow us to store the state/geometry safely as text.

> Doesn't work...

Hard to say what you should try next without some more info. Did you also base64-decode the data when reading it from the config? ```QByteArray::fromBase64``` is the opposite of ```QByteArray::toBase64```.